src/AutoRest.CSharp/Common/Output/Expressions/ValueExpressions/ValueExpression.cs

```diff
@@ -37,10 +40,10 @@ public ValueExpression Invoke(string methodName, IReadOnlyList<ValueExpression>
             => new InvokeInstanceMethodExpression(this, methodName, arguments, null, false);
 
         public ValueExpression Invoke(MethodSignature method)
-            => new InvokeInstanceMethodExpression(this, method.Name, method.Parameters.Select(p => (ValueExpression)p).ToList(), null, false);
+            => new InvokeInstanceMethodExpression(this, method.Name, method.Parameters.Select(p => (ValueExpression)p).ToList(), null, method.Modifiers.HasFlag(MethodSignatureModifiers.Async));
```

Comment:
This won't work for the pageables (they aren't async).
I think we can introduce `IsAsync` property in `MethodSignature` by extracting this line:
https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Common/Output/Models/MethodSignature.cs#L24

Reply:
I do not think so it will not work on pageables, I made this change because I am getting issues on async pageable operation invocations.
The "async pageable" operations are not async, therefore they do not have the `async` keyword on their signature, therefore we do not need to add the `await` keyword when invoking them.